PublicSquare/Payments/Api/Authenticated/Payments.php

```diff
@@ -56,10 +124,44 @@ protected function getPath(): string
     protected function getMethod(): string
     {
         return \Laminas\Http\Request::METHOD_POST;
-    }//end getMethod()
+    } //end getMethod()
 
-    public function getData(): array
+    protected function validateResponse(mixed $data): bool
     {
-        return $this->data;
-    }//end getData()
-}//end class
+        if (in_array($data["status"], ["succeeded", "requires_capture"])) {
```

Comment:
are we sure that "status" will exist in the array. 
perhaps do something like this: 
$status = !empty($data["status"])? $data["status"] : "";